native/src/lib.rs

```diff
@@ -81,11 +85,107 @@ declare_types! {
                                 None)
                 .expect("Query failed");
 
-            Ok(try!(JsString::new_or_throw(scope, &results.len().to_string()[..])).upcast())
+            if let &QueryResults::Scalar(Some(TypedValue::Keyword(ref rc))) = results {
```

Comment:
Rather than a chain of `if let`:

```
use QueryResults::*;
match results {
    &Scalar(None) => {
        // Didn't return a result. JS `undefined`?
    },
    &Scalar(Some(v)) => {
        process_typed_value(scope, v)
    },
    &Coll(vals) => { …
```